### ISSUE TITLE
Please provide this via F-Droid.

### ISSUE THREAD

**Issue by rokejulianlockhart** [recloudstream/cloudstream#336]
### Describe your suggested feature

My rationale is available at https://github.com/Tadashi-Hikari/Sapphire/issues/2#issue-1228766021.

### Other details

Reproducible builds allow you to use your signature when it compiles at F-Droid, thereby allowing updates from any source to be compatible.

https://github.com/recloudstream/cloudstream#features demonatrates that not much work should be necessary, if any.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by LagradOst** [comment]
they use own sig, so no I cant switch without fucking it up for them nor do I want a public keycode

**Comment by rokejulianlockhart** [comment]
@LagradOst, I have stated that F-Droid now supports “reproducible builds” which means that they support developer-provided signatures.

Right, @IzzyOnDroid?

**Comment by IzzySoft** [comment]
> F-Droid now supports “reproducible builds” which means that they support developer-provided signatures

for years already, yes. But that requires the app to build reproducibly – i.e. your app stripped of its signature is binary-identical to the app built by F-Droid.

**Comment by DarkCrypt** [comment]
2-1-2023: The number of apps published with reproducible builds on F-Droid = 60.

I agree with what LagradOst and IzzySoft are saying. I'm all about privacy and I'd want to keep my app signature in my own hands instead of handing it out everywhere. I'm also a huge fan of FOSS apps and where to find them.

This app isn't on F-Droid but there's a stable version on IzzyOnDroid. You could put Izzy's repo in your F-Droid client and get updates from there. Or you could do what I do and use [Neo Store](https://github.com/NeoApplications/Neo-Store) as an F-Droid client which has multiple repos to choose from. Besides the numerous clients, this app also provides in app updates.

**Comment by IzzySoft** [comment]
> The number of apps published with reproducible builds on F-Droid = 60.

And steadily growing, as the process got easier (mostly thanks to obfusk investing a lot of time into it and even providing tools for it). 10 of those 60 were added in January alone, more to come. And exactly good for the reason of "best of 2 worlds": the app signed by the original dev but checked by the F-Droid team. So who now wants to claim either of the two put something in before signing? No way until both ends would do that. So whomever of the two you trust, you just got another proof. All sides win.

A slight win with my repo: you get some additional checks (malware & library scanners). So you indeed could start with that. Once it's established at F-Droid and being reproducible, you can simply switch there with an update, as signatures would match.

**Comment by DarkCrypt** [comment]
> And steadily growing, as the process got easier (mostly thanks to obfusk investing a lot of time into it and even providing tools for it). 10 of those 60 were added in January alone, more to come. And exactly good for the reason of "best of 2 worlds": the app signed by the original dev but checked by the F-Droid team. So who now wants to claim either of the two put something in before signing? No way until both ends would do that. So whomever of the two you trust, you just got another proof. All sides win.

Right, there will be many more to come as they are encouraging all devs to do this. I didn't understand reproducible builds all that well but your comment made it much more clear. Thank you for your explanation.

**Comment by beedell-roke** [comment]
> [`#issuecomment-1405501402`](https://github.com/recloudstream/cloudstream/issues/336#issuecomment-1405501402)

It's at least currently available at [`apt.izzysoft.de/fdroid/index/apk/com.lagradost.cloudstream3`](https://apt.izzysoft.de/fdroid/index/apk/com.lagradost.cloudstream3).

**Comment by IzzySoft** [comment]
Since 2021-08-26, yupp – so for almost 3 years now :smiley:

**Comment by cultx7191** [comment]
Not anymore sadly

**Comment by IzzySoft** [comment]
@cultx7191 see https://github.com/LagradOst/QuickNovel/issues/223#issuecomment-2289503318 – it sadly had to be removed.

**Comment by cultx7191** [comment]
@IzzySoft understood, though it's still a bummer since it's such a great app. Obtainium exists at least

**Comment by IzzySoft** [comment]
Yeah. I find it unfortunate, too; Obtainium exists, but doesn't run the extra checks we do at IzzyOnDroid. But we had no choice. We can reactivate the apps anytime should the situation change, no grudges hold or something. Decision was out of mutual respect, not out of a fight :wink:

**Comment by Korb** [comment]
In my F-Droid app, I see two versions of the CloudStream app - stable and beta.

![Image](https://github.com/user-attachments/assets/ab7d2c20-cc2d-48bc-8c20-10888c817c28)

With this in mind, is it worth insisting on an app being located in the F-Droid directory when the app can be found, installed, and automatically updated through an alternative repository?

**Comment by beedell-roke** [comment]
@Korb, yes, because “C10udburst”'s repository does not provide the guarantees that the primary repository does (compilation from source, etcetera). If you're going to trust any old repository, you might as well merely retrieve the binaries from Obtanium. It'd be quicker and more reliable.

**Comment by jxmesth** [comment]
@RokeJulianLockhart, some good points made here.

@fire-light42, is this being worked on?

**Comment by spidy2356** [comment]
@RokeJulianLockhart how to add cs beta on obtainium?
For me obtainium can't detect latest beta versions..how to solve?

Or any other solution would be appreciated

**Comment by beedell-roke** [comment]
@spidy2356, best to ask at Obtanium's Discussion section, but:

1. <img width="1144" height="1648" alt="Image" src="https://github.com/user-attachments/assets/d409b142-be2e-4124-a915-92b95e674c88" />

1. <img width="1158" height="2056" alt="Image" src="https://github.com/user-attachments/assets/9d1a9646-f6c9-454e-9983-e30149b1d4a3" />

1. <img width="1034" height="949" alt="Image" src="https://github.com/user-attachments/assets/fa295d7a-c806-4102-b8f8-dfbee9783090" />

To summarise, include pre-releases.
